py/runtime.h

```diff
@@ -158,6 +160,13 @@ NORETURN void mp_raise_NotImplementedError(const char *msg);
 NORETURN void mp_raise_OSError(int errno_);
 NORETURN void mp_raise_recursion_depth(void);
 
+mp_obj_t mp_raise_o(mp_obj_t exc);
```

Comment:
I think these should be macros instead. Or at least, macro variants provided:

~~~
#define MP_RAISE_O(exc) { return mp_raise_o(exc); }
~~~

Reply:
> If it's important to have them as macros/inline functions

So, I'm interested in starting to adopt these sooner rather than later. Then, having that as a *macro* is important. For example, MP_RAISE_NEW() could be defined as:

~~~
#if USE_NEW_EXC
#define MP_RAISE_NEW(exc) { return mp_raise_o(exc); }
#else
#define MP_RAISE_NEW(exc) nlr_raise(exc);
#endif
~~~

And gradual conversion can be started right away. Again, exact naming is not important, the argument for incremental conversion started "soon". Whereas waiting for "once everything was converted" is a separate, always-bitrotting patch may take years.